### PR TITLE
Direct-XIP with BT OTA DFU alignment

### PR DESCRIPTION
**Pull request by mkapala-nordic**

### CODE REVIEW

File: samples/common/mcumgr_bt_ota_dfu/Kconfig [hunk 1]
```diff
@@ -19,7 +19,12 @@ config NCS_SAMPLE_MCUMGR_BT_OTA_DFU
 	imply FLASH_MAP
 	imply FLASH
 	imply MCUMGR_GRP_IMG
+	imply MCUMGR_GRP_IMG_REJECT_DIRECT_XIP_MISMATCHED_SLOT if \
+		NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT
+	imply MCUMGR_GRP_IMG_DIRECT_UPLOAD if \
+		NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT
 	imply MCUMGR_GRP_OS
+	imply MCUMGR_GRP_OS_BOOTLOADER_INFO
```
Comment (mkapala-nordic) [inline samples/common/mcumgr_bt_ota_dfu/Kconfig:27]: Comment from @kapi-no from previous PR:

@philips77, can we have the Bootloader info command available in all MCUboot bootloader modes?

File: samples/common/mcumgr_bt_ota_dfu/Kconfig [hunk 2]
```diff
@@ -134,3 +139,8 @@ config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_VALIDATION
 	  Enable this option to validate the MCUmgr OTA DFU configuration at the
 	  build time. If the project configuration does not match predefined
 	  defaults, the build system issues a warning for each misaligned item.
+
+config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT
+	bool
+	default y if MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP
+	default y if MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT
```
Comment (kapi-no) [inline samples/common/mcumgr_bt_ota_dfu/Kconfig:146]: you can move implies here without any conditions to achieve the same effect

Comment (mkapala-nordic) [reply samples/common/mcumgr_bt_ota_dfu/Kconfig:142]: done

**Review by kapi-no — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

File: samples/common/mcumgr_bt_ota_dfu/Kconfig (hunk 1 above)
Comment (philips77) [reply samples/common/mcumgr_bt_ota_dfu/Kconfig:23]: I don't think XIP is possible for nRF53 at all.

**Review by philips77 — APPROVED** [review]

**Pull request merged**